public/app/plugins/datasource/grafana-pyroscope-datasource/QueryEditor/QueryEditor.tsx

```diff
@@ -27,8 +27,15 @@ export function QueryEditor(props: Props) {
     onRunQuery();
   }
 
+  // Need to reference the query as otherwise when the label selector is changed, only the initial value
+  // of the query is passed into the LabelsEditor (onChange) which renders the CodeEditor for monaco.
+  // The above needs to have a ref to the query so it can get the latest value.
+  const queryRef = useRef(query);
+  queryRef.current = query;
```

Comment:
Nice yeah this is what I had in mind but I also think we can move the definition of the ref inside the `useLabelSelector`. That way the whole thing is encapsulated and a single "unit" so to say. It shouldn't change the functionality as the variable capturing happens inside `useLabelSelector` and the queryRef now isn't needed anywhere else anyway.

Reply:
Awh my God, yeah makes sense, thanks for the re-review!